static/js/terminal.js

```diff
@@ -180,9 +180,14 @@ function filterProcedures() {
 }
 function showProcedure() {
     function displayProcedure(data){
-        let a = data[0];
-        term.write(atob(a.test));
-        input = atob(a.test);
+        for (let ab of data) {
+            let agent = $('#session-id option:selected');
+            if (ab.platform === agent.data("platform") && agent.data("executors").includes(ab.executor)) {
+                term.write(atob(ab.test));
+                input = atob(ab.test);
+                break;
```

Comment:
it seems that we should print something if there is no match

Reply:
or is the data filtered before getting here, so there's really no chance of that?

Reply:
I haven't dug deep enough to see if the data is getting filtered. I can poke

Reply:
@privateducky Looks like it is filtered initially:
```javascript
restRequest('POST', {"paw": $('#session-id option:selected').data('paw')}, getAbilitiesCallback, '/plugin/manx/ability');
```